utils/generate_snowball.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+
+"""Refresh and update language stemming data from the Snowball project."""
+
+# /// script
+# requires-python = ">=3.11"
+# dependencies = [
+#     "requests>=2.30",
+# ]
+# ///
+
+from __future__ import annotations
+
+import shutil
+import subprocess
+import sys
+import tarfile
+import tempfile
+from io import BytesIO
+from pathlib import Path
+
+import requests
+
+SNOWBALL_VERSION = '3.0.1'
+SNOWBALL_URL = f'https://github.com/snowballstem/snowball/archive/refs/tags/v{SNOWBALL_VERSION}.tar.gz'
```

Comment:
Although I recognize that this is a utility script and therefore any changes introduced by it are likely to go through code review / etc, I'm generally fairly strongly in favour of pinning checksums along with download of static content.

In other words: because we know that we're downloading v3.0.1 of `snowball` here, I think we could/should assert that the SHA256sum of the resulting download matches an expected value.

There is a small chance that GitHub GZ compression might change in future, as they have [once before](https://github.blog/changelog/2023-01-30-git-archive-checksums-may-change/) -- but such events should be rare, so I _don't_ think it would be worth being clever and trying to checksum the `.tar` or otherwise determine the inner contents of the archive.

```suggestion
SNOWBALL_URL = f'https://github.com/snowballstem/snowball/archive/refs/tags/v{SNOWBALL_VERSION}.tar.gz'
SNOWBALL_SHA256 = '80ac10ce40dc4fcfbfed8d085c457b5613da0e86a73611a3d5527d044a142d60'
```

Reply:
Done